src/components/app/App.test.tsx

```diff
@@ -0,0 +1,290 @@
+import { render, screen, act, waitFor, fireEvent } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom'
+import * as httpGetModule from '../../utils/httpGet';
+import { httpGet } from '../../utils/httpGet';
+import App from './App';
+import { PAGE_NUMBER_DEFAULT } from '../../utils/constants';
+import type { ISectionsResponseResults, ISectionsResponse, ISearchResponseResults, ISearchResponse } from '../../types/types';
+
+describe(`App`, () => {
+    beforeAll(() => {
+        window.scrollTo = jest.fn();
+    })
+
+    beforeEach(() => {
+        jest.useFakeTimers()
+    });
+
+    afterEach(() => {
+        jest.runOnlyPendingTimers()
+        jest.useRealTimers()
+    });
+
+    const mockedSectionsResponseResultsList: ISectionsResponseResults[] = Array.from(new Array(20).fill(null).map((_, idx) => (
+        {
+            id: `test-section-id-${idx + 1}`,
+            webTitle: `test-section-webTitle-${idx + 1}`,
+            webUrl: 'test-section-webUrl',
+        }
+    )));
+
+    const mockedSectionsResponse: ISectionsResponse = {
+        results: mockedSectionsResponseResultsList,
+    };
+
+    const mockedSearchResponseResultsList: ISearchResponseResults[] = Array.from(new Array(10).fill(null).map((_, idx) => (
+        {
+            id: `test-article-id-${idx + 1}`,
+            type: 'test-article-type',
+            sectionId: 'test-article-sectionId',
+            sectionName: 'test-article-sectionName',
+            webPublicationDate: new Date('2023-01-01').toString(),
+            webTitle: 'test-article-webTitle',
+            webUrl: 'test-article-webUrl',
+        }
+    )));
+
+    const mockedSearchResponse: ISearchResponse = {
+        total: 101,
+        startIndex: 1,
+        pageSize: 10,
+        currentPage: 1,
+        pages: 11,
+        results: mockedSearchResponseResultsList,
+    };
+
+    describe(`requests failed`, () => {
+        it(`should render "sections" and "search" loading error indication`, async () => {
```

Comment:
Okey so here I see two issues:
- for displaying sections and errors related to getting sections <Sidebar /> component is responsible
- for articles on other hand you did not create separate component, which actually would be nice to encapsulate error display in it like you are doing this in Sidebar case
That way if error displaying will be "delegated" to those two components it should be tested there. In App you only care about displaying those two components - or even not that - you care more about providing proper data to those components, fetching this data, and those components care about displaying themselves.